@AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/core/fn_coreUpdate.sqf

```diff
@@ -0,0 +1,225 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_sfmplusUpdate
+
+Description:
+	
+Parameters:
+	_heli - The apache helicopter to get information from [Unit].
+
+Returns:
+
+
+Examples:
+
+
+Author:
+	BradMick
+---------------------------------------------------------------------------- */
+params ["_heli"];
+
+private _deltaTime = ["sfmplus_deltaTime"] call BIS_fnc_deltaTime;
+
+//Input
+[_heli] call fza_sfmplus_fnc_getInput;
+
+//Weight
+private _emptyMass      = _heli getVariable "fza_sfmplus_emptyMass";
+private _maxTotFuelMass = _heli getVariable "fza_sfmplus_maxTotFuelMass";
+private _fwdFuelMass    = [_heli] call fza_sfmplus_fnc_fuelSet select 0;
+private _aftFuelMass    = [_heli] call fza_sfmplus_fnc_fuelSet select 1;
+
+//Engines
+[_heli, _deltaTime] call fza_sfmplus_fnc_engineController;
+
+//Fuel
+private _apuFF  = 0;
+private _eng1FF = _heli getVariable "fza_sfmplus_engFF" select 0;
+private _eng2FF = _heli getVariable "fza_sfmplus_engFF" select 1;
+private _curFuelFlow = 0;
+
+if (_heli animationphase "plt_apu" > 0.5) then {
+	_apuFF = 0.0220;	//175pph
+};
+_curFuelFlow    = (_apuFF + _eng1FF + _eng2FF) * _deltaTime;
+
+private _totFuelMass  = _fwdFuelMass + _aftFuelMass;
+_totFuelMass          = _totFuelMass - _curFuelFlow;
+private _armaFuelFrac = _totFuelMass / _maxTotFuelMass;
+_heli setFuel _armaFuelFrac;
+
+//Pylons
+private _pylonMass = 0;
+{
+	_x params ["_magName","", "_magAmmo"];
+	private _magConfig    = configFile >> "cfgMagazines" >> _magName;
+	private _magMaxWeight = getNumber (_magConfig >> "weight");
+	private _magMaxAmmo   = getNumber (_magConfig >> "count");
+	_pylonMass = _pylonMass + linearConversion [0, _magMaxAmmo, _magAmmo, 0, _magMaxWeight];
+} foreach magazinesAllTurrets _heli;
+
+private _curMass = _emptyMass + _totFuelMass + _pylonMass;
+_heli setMass _curMass;
+
+//Damage
+[_heli, _deltaTime] call fza_sfmplus_fnc_damageApply;
+
+//Stabilator
+if(fza_ah64_sfmPlusStabilatorEnabled) then {
+	[_heli, _deltaTime] call fza_sfmplus_fnc_aeroStabilator;
+};
+/*
+hintsilent format ["v0.3
```

Comment:
For now, if you have a load of stuff you need to keep for testing, just throw it in a gist for now